src/AppKit/NSButton.cs

```diff
@@ -37,7 +37,14 @@
 namespace AppKit {
 
 	public partial class NSButton {
-		NSActionDispatcher? dispatcher;
+		NSObject? dispatcher;
```

Comment:
I've noticed this pattern in a few other pr's too, but why is there a separate dispatcher NSObject? Looking at the [docs](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/set) it makes sense to use this pattern if you want to keep the property variable private and use it for private-backing, but in this case what is the benefit to doing this rather than just using the auto-implemented property?

Reply:
> in this case what is the benefit to doing this rather than just using the auto-implemented property

In this case the purpose of the property setter is have a side effect: call MarkDirty. That's not possible for auto-implemented properties.